src/ws/transport.rs

```diff
@@ -425,14 +424,18 @@ async fn per_connection_task(
                             crate::common::Version::V2,
                         ))
                         .expect("valid JSON; qed");
+
+                        // deserialization failed and the client is not alive then close the connection.
                         if let Err(e) = sender.send_text(&response).await {
                             log::warn!(
                                 "Failed to send: {:?} over WebSocket transport with error: {:?}",
                                 response,
                                 e
                             );
+                            return pending_requests;
+                        } else {
+                            continue;
```

Comment:
Instead of returning here i.e, closing the connection, continue the loop, and spawn new futures to wait for.